### ISSUE TITLE
SIGSEGV during SSL.freeSSL when multiple instances are deployed

### ISSUE THREAD

**Issue by conet** [eclipse-vertx/vert.x#5124]
### Questions

This is related to this netty-tcnative issue https://github.com/netty/netty-tcnative/issues/842

### Version

4.5.3 and older.

### Context

We noticed the crash on high traffic servers with a high SSL connection creation/destruction rate. Today we got some DoS like traffic on some instances and they crash almost immediately after restart. I realized that the application was deploying multiple verticles. Once we switch to a single verticle per process the JVM no longer crashes. Another way to prevent the crash is to use `JdkSSLEngineOptions` instead of `OpenSSLEngineOptions`.

### Do you have a reproducer?

I don't have a reproducer.

### Extra

[hs_err_pid389199.log](https://github.com/eclipse-vertx/vert.x/files/14324541/hs_err_pid389199.log)

**Comment by conet** [comment]
Sorry, I spoke too soon, it also happens when only one verticle is deployed.

**Comment by conet** [comment]
Was this fixed by https://github.com/netty/netty/security/advisories/GHSA-4g8c-wm8x-jfhw?

**Comment by vietj** [comment]
> Was this fixed by [GHSA-4g8c-wm8x-jfhw](https://github.com/netty/netty/security/advisories/GHSA-4g8c-wm8x-jfhw)?

it is not clear, it should be determined, that being said you should upgrade to the most recent version of Vert.x that fixes it

**Comment by conet** [comment]
Yes, I'm testing that, I will observe the new version with openssl enabled for a month or so.
